test/WebSites/ValidationWebSite/Controllers/ModelMetadataTypeAttributeController.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc;
+using ValidationWebSite.ViewModels;
+
+namespace ValidationWebSite.Controllers
+{
+    public class ModelMetadataTypeValidationController : Controller
+    {
+        [HttpPost]
+        public object ValidateProductViewModelInclMetadata([FromBody] ProductViewModel product)
+        {
+            return CreateValidationDictionary();
+        }
+
+        [HttpPost]
+        public object ValidateSoftwareViewModelInclMetadata([FromBody] SoftwareViewModel software)
+        {
+            return CreateValidationDictionary();
+        }
+
+        [HttpPost]
+        public object TryValidateModelProductViewModelWithErrorInParameter(int id, [FromBody] ProductViewModel product)
+        {
+            if (id == 0)
+            {
+                ModelState["id"].Errors.Clear();
+            }
+            TryValidateModel(product);
+
+            return CreateValidationDictionary();
+        }
+
+        [HttpGet]
+        public object TryValidateModelSoftwareViewModelNoPrefix()
+        {
+            var softwareVm = new SoftwareViewModel
+            {
+                Category = "Technology",
+                CompanyName = "Microsoft",
+                Contact = "4258393231",
+                Country = "UK",
+                DatePurchased = new DateTime(10, 10, 10),
+                Price = 110,
+                Version = "2"
+            };
+
+            TryValidateModel(softwareVm);
+
+            return CreateValidationDictionary();
+        }
+
+        [HttpGet]
+        public object TryValidateModelValidModelNoPrefix()
```

Comment:
both this action and `TryValidateModelSoftwareViewModelNoPrefix` call `TryValidateModel()` without a prefix.  either change this one to use a prefix or add another action (and corresponding test).